### PR TITLE
Implements new T-Rays

### PR DESCRIPTION
**Pull request by MisterLayne**
Tested, working, all good.

Edit: Tech levels are VERY much open to changing, if anyone wants to make suggestions?

### CODE REVIEW

**Comment by mistyLuminescence** [comment]
> req_tech = list(TECH_MAGNET = 6, TECH_ENGINEERING = 7, TECH_MATERIAL = 7)

Unobtainable without technomancer core. Try req_tech of...

For the upgraded/CE version: (TECH_MAGNET = 3, TECH_ENGINEERING = 4, TECH_MATERIAL = 2)
For the advanced/RnD version: (TECH_MAGNET = 6, TECH_ENGINEERING = 6, TECH_MATERIAL = 6)

File: code/modules/research/designs.dm [hunk 1]
```diff
@@ -853,6 +853,25 @@ other types of metals and chemistry for reagents).
...
+	build_path = /obj/item/device/t_scanner/upgraded
+	sort_string = "VASSA"
+
+
+/datum/design/item/device/t_scanner_adv
+	name = "Advanced T-ray Scanner"
+	desc = "An advanced version of the terahertz-ray emitter and scanner used to detect underfloor objects such as cables and pipes."
+	id = "advancedtscanner"
+	req_tech = list(TECH_MAGNET = 3, TECH_ENGINEERING = 4, TECH_MATERIAL = 2)
+	materials = list(TECH_MAGNET = 6, TECH_ENGINEERING = 6, TECH_MATERIAL = 6)
```
Comment (Atermonera) [inline code/modules/research/designs.dm]: Those aren't materials

Comment (MisterLayne) [reply code/modules/research/designs.dm]: >->
You might just be right.

Comment (MisterLayne) [reply code/modules/research/designs.dm]: That's why I got a travis issue, hmf.

**Comment by Anewbe** [comment]
Tested, etc?

**Pull request merged**